src/services/filters-parser.js

```diff
@@ -0,0 +1,186 @@
+import _ from 'lodash';
+import Operators from '../utils/operators';
+import OperatorDateIntervalParser from './operator-date-interval-parser';
+import { NoMatchingOperatorError, InvalidFiltersFormatError } from './errors';
+
+function FiltersParser(filtersString, timezone, options) {
+  this.OPERATORS = new Operators(options);
+  this.operatorDateIntervalParser = new OperatorDateIntervalParser(timezone, options);
+
+  try {
+    this.filters = filtersString ? JSON.parse(filtersString) : null;
+  } catch (error) {
+    throw new InvalidFiltersFormatError('Invalid filters JSON format');
+  }
+
+  this.perform = () => {
+    if (!this.filters) return null;
+
+    return this.formatAggregation(this.filters);
+  };
+
+  this.formatAggregation = (node) => {
+    if (_.isEmpty(node)) throw new InvalidFiltersFormatError('Empty condition in filter');
```

Comment:
What if it's the wrong format, something like: `{ pwned: 'dead' }`

Reply:
It's already handled: it will be treated as a non-empty flat condition then the operator will be parsed and not recognised => error raised